### PR TITLE
Feature/task2 components and linter

### PR DESCRIPTION
**Pull request by txlyzov**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
+  "root": true,
+  "plugins": [
+    "@typescript-eslint",
+    "import"
+  ],
+  "parser": "@typescript-eslint/parser",
+  // "parser": "vue-eslint-parser",
```
Comment (lgordienko) [inline .eslintrc.json]: Please remove all comments for .eslintrc file.
I think they cant help with further development.

File: src/app/app.component.html [hunk 2]
```diff
@@ -21,12 +20,21 @@ <h3 class="px-3 mb-0 text-secondary">Test Angular application</h3>
...
+            <button type="button" class="btn navbar-btn btn-warning" routerLink="/register">Sign-up</button>
+        </div>
         <hr class="mt-3 mb-2 dropdown-divider" style="border-color:white;">
     </div>
 </header>
 
+<ng-template #loggedOut>
+    <h4 class="nav-item mb-auto">
+        <a href="#" class="ps-0 pb-0 nav-link text-warning" routerLink="/images-control">
+            {{User}}</a>
```
Comment (lgordienko) [inline src/app/app.component.html]: Please move end tag to the next line

File: src/app/app.component.html [hunk 3]
```diff
@@ -21,12 +20,21 @@ <h3 class="px-3 mb-0 text-secondary">Test Angular application</h3>
...
+    </h4>>
```
Comment (lgordienko) [inline src/app/app.component.html]: Please review that line. Is it ok to have doubled ">" symbol?

File: src/app/components/image-card/image-card.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="card p-1 m-1">
+
+    <app-image class="m-auto" [image]=image></app-image>
+
+    <div *ngIf="!isOnlyDisplaying">
+        <div class="d-flex justify-content-between pt-1 mt-1 border-top border-secondary">
+            <button type="button" class="btn btn-outline-dark btn-sm py-0"><i class="bi bi-eye"></i></button>
```
Comment (lgordienko) [inline src/app/components/image-card/image-card.component.html]: Please move <button> block content (I mean <i class ... /i>) to the next line.
And fix same cases in file.

File: src/app/components/image-card/image-card.component.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+// import { EventEmitter } from 'stream';
```
Comment (lgordienko) [inline src/app/components/image-card/image-card.component.ts]: Is this comment required?

File: src/app/types/any-for-moda-data.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line
+export type AnyForModalData = any;
```
Comment (lgordienko) [inline src/app/types/any-for-moda-data.ts]: I'm not sure that variable name like AnyForModalData is quiet common and understandable.
Please rename it to "Any", "CustomAny", "AnyType", smth like that.
Also please review filename.

**Comment by lgordienko** [comment]
The file src/app/components/delete-image-modal/delete-image-modal.component.css is empty. Maybe it's better to delete it?

File: src/app/components/gallery-elements-bar/gallery-elements-bar.component.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<div class="mx-4 mt-2 pb-1 border-bottom bd-highlight border-dark d-flex justify-content-between">
+
+    <div class="p-2 mx-3 bd-highlight">
+        <input type="search" class="form-control form-control-dark" placeholder="Search..." aria-label="Search">
+    </div>
+
+    <div *ngIf="isAvaliable" class="p-2 me-auto bd-highlight">
+        <app-select></app-select>
```
Comment (lgordienko) [inline src/app/components/gallery-elements-bar/gallery-elements-bar.component.html:8]: Please review (throw all pr) is it possible to replace such cases to <app-select />.

File: src/app/pages/user-images/user-images.component.css [hunk 8]
```diff
@@ -1,49 +1,4 @@
...
-
-.image-text {
-    top: 15%;
-    position: relative;
-    color: black;
-}
-
-i {
-    font-size: 1.4rem;
-}
```
Comment (lgordienko) [inline src/app/pages/user-images/user-images.component.css:49]: I'm not sure that there is a newline in this file (and other .css files). Could you review that and update if required?

Comment (txlyzov) [reply src/app/pages/user-images/user-images.component.css:49]: Already done.

File: src/app/components/gallery-elements-bar/gallery-elements-bar.component.html (hunk 7 above)
Comment (txlyzov) [reply src/app/components/gallery-elements-bar/gallery-elements-bar.component.html:8]: No way to do this.

**Review by lgordienko — APPROVED** [review]

**Comment by txlyzov** [comment]
Divided by components,rebulded html forms and setting up ESlint.

**Pull request merged**